src/olympia/amo/fields.py

```diff
@@ -38,7 +39,17 @@ def __call__(self, value):
 
 
 class HttpHttpsOnlyURLField(fields.URLField):
-    default_validators = [URLValidatorBackport(schemes=('http', 'https'))]
+    default_validators = [
+        URLValidatorBackport(schemes=('http', 'https')),
+        # Reject AMO URLs, see:
+        # https://github.com/mozilla/addons-server/issues/9012
+        RegexValidator(
+            regex=r'%s' % settings.SERVICES_DOMAIN,
+            message=_('Linking to AMO is forbidden.'),
```

Comment:
@jvillalobos can you provide some copy for this error message? "Forbidden" reads too strongly imo.

Reply:
"This field can only be used to link to external websites. URLs on addons.mozilla.org are not allowed."
If it's too long, the first sentence should be enough.